pkg/protocol/msg_tx.go

```diff
@@ -15,6 +15,15 @@ type MsgTx struct {
 	transaction.Transaction
 }
 
+func NewMsgTxFromHex(hex string) (MsgTx, error) {
+	tx, err := transaction.NewTxFromHex(hex)
+	if err != nil {
+		return MsgTx{}, err
+	}
+
+	return MsgTx{*tx}, nil
+}
```

Comment:
shouldn't this enforce all protocol checks to make sure the tx can be broadcasted correctly? I guess we likely we don't want a node able to spam bad transactions to the network :)

Reply:
yep u're right! However I can't do all the protocol checks. For instance `missing-inputs` need a kind of indexing. Only things I can do is "sanity checks" like if the tx is decodable or version, locktime etc... but if a malicious guy wants to SPAM the network, it is possible. (but full node peers should ban the spammer!)

Reply:
Will keep it simple for now? we do not run a mempool anyway?

Reply:
I would keep it simple yes, `neutrino` do the same thing and implement a `Broadcaster` service for optimization and checks. 

We do not run a mempool yet, but we definitively need it (not the whole mempools but we can apply filters on it)

Reply:
Ok so things like verification of inputs, inputs' signatures and blinding correctness will all be part of the Broadcaster service?

Reply:
in general for privacy is fine to use an explorer per broadcasting, instead own node.

I think we can implement broadcaster later?

Reply:
yes we need firstly a clean way to discover peers (now Broadcaster does not make sense with 1 or 2 peers I think).

> Ok so things like verification of inputs, inputs' signatures and blinding correctness will all be part of the Broadcaster service?

Not sure if this should be the role of the SPV node, we do not "build blocks" :thinking: 
Do u know how bitcoin SPV nodes deal with broadcasting, tx verification?

Reply:
AFAIK [neutrino](https://github.com/lightninglabs/neutrino) does not keep a local mempool or relay new transaction any way